File: Selenium/Project/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    # Navigate to the webpage
    driver.get("http://alchemy.hguy.co/orangehrm")

    # Find and fill in the username and password fields
    username = driver.find_element(By.XPATH, "//input[@id = 'txtUsername']")
    password = driver.find_element(By.XPATH, "//input[@id = 'txtPassword']")
    username.send_keys("orange")
    password.send_keys("orangepassword123")

    # Click the login button
    driver.find_element(By.XPATH, "//input[@id = 'btnLogin']").click()

    # Implicitly wait for 10 seconds
    driver.implicitly_wait(10)

    # Find the PIM element and click on it
    PIM = driver.find_element(By.XPATH, "//*[@id='menu_pim_viewPimModule']")
    PIM.click()

    # Add new employee
    addBtn = driver.find_element(By.XPATH, "//*[@id='btnAdd']")
    addBtn.click()

    firstName = driver.find_element(By.XPATH, "//*[@id='firstName']")
    lastName = driver.find_element(By.XPATH, "//*[@id='lastName']")
    firstName.send_keys("Orange")
    lastName.send_keys("OrangeHRM")

    # Save new employee
    btnSave = driver.find_element(By.XPATH, "//*[@id='btnSave']")
    btnSave.click()

    # View employee list
    viewEmployee = driver.find_element(By.XPATH, "//*[@id='menu_pim_viewEmployeeList']")
    viewEmployee.click()

    # Search by Name
    searchName = driver.find_element(By.XPATH, "//*[@id='empsearch_employee_name_empName']")
    searchName.send_keys("Orange")

    searchBtn = driver.find_element(By.XPATH, "//*[@id='searchBtn']")
    searchBtn.click()
    driver.quit()

File: Selenium/Project/Activity4.java
package Projects_HRM;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.concurrent.TimeUnit;

public class Activity4 {

    static WebDriver driver;

    @BeforeClass
    public static void setup() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        driver.get("http://alchemy.hguy.co/orangehrm");
    }

    @Test
    public void verify() {

        WebElement userName = driver.findElement(By.xpath("//*[@id='txtUsername']"));
        WebElement password = driver.findElement(By.xpath("//*[@id='txtPassword']"));
        WebElement loginBtn = driver.findElement(By.xpath("//*[@id='btnLogin']"));
        //--------- Login
        userName.sendKeys("orange");
        password.sendKeys("orangepassword123");
        loginBtn.click();

        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
        //---------- CLick PIM
        WebElement PIM = driver.findElement(By.xpath("//*[@id='menu_pim_viewPimModule']"));

        PIM.click();
        //---------Add new
        WebElement addBtn = driver.findElement(By.xpath("//*[@id='btnAdd']"));
        addBtn.click();
        WebElement firstName = driver.findElement(By.xpath("//*[@id='firstName']"));
        WebElement lastName = driver.findElement(By.xpath("//*[@id='lastName']"));
        firstName.sendKeys("Orange");
        lastName.sendKeys("OrangeHRM");
        // Save new
        WebElement btnSave = driver.findElement(By.xpath("//*[@id='btnSave']"));
        btnSave.click();
        //-------- view employee
        WebElement viewEmployee = driver.findElement(By.xpath("//*[@id='menu_pim_viewEmployeeList']"));
        viewEmployee.click();

        //------- search by Name
        WebElement searchName = driver.findElement(By.xpath("//*[@id='empsearch_employee_name_empName']"));
        searchName.sendKeys("Orange");
        //WebDriverWait  = new WebDriverWait(driver, Duration.ofSeconds(10));
        WebElement searchBtn = driver.findElement(By.xpath("//*[@id='searchBtn']"));
        searchBtn.click();

    }

    @AfterClass
    public static void afterMethod() {
        //Close the browser
        driver.quit();
    }

}